### PR TITLE
Added a splash screen with DB version check.

### LINKED ISSUE #30: Check DB compatibility on app startup
When application starts a compatibility check will be made to a version saved in the DB and if the version does not match it will transfer the user to the app store, and would not allow the application to start.

### LINKED ISSUE #90: Checkbox enabled in view inspection
When watching an inspection made to a boat, the checkboxes should be read-only.

### PR DESCRIPTION
**Pull request by aayaffe**
Closed #30
Closed #90
Refactoring

### CODE REVIEW

File: app/src/main/java/in/avimarine/boatangels/activities/SplashActivity.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      } else {
+        Intent intent = new Intent(this, MainActivity.class);
+        startActivity(intent);
+        finish();
+      }
+    });
+  }
+
+  private void alertOnUnsupportedVersion() {
+    Log.d(TAG, "Version not supported, starting dialog");
```
Comment (Pazit) [inline app/src/main/java/in/avimarine/boatangels/activities/SplashActivity.java]: I would add a printing of current installed version number and of the supported one.

File: app/src/main/java/in/avimarine/boatangels/fragments/MyBoatFragment.java [hunk 2]
```diff
@@ -81,18 +71,12 @@ public MyBoatFragment() {
   /**
    * Use this factory method to create a new instance of
    * this fragment using the provided parameters.
-   *
-   * @param param1 Parameter 1.
-   * @param param2 Parameter 2.
+
    * @return A new instance of fragment MyBoatFragment.
    */
   // TODO: Rename and change types and number of parameters
```
Comment (Pazit) [inline app/src/main/java/in/avimarine/boatangels/fragments/MyBoatFragment.java]: TODELETE (all comments related to "Param"s)

File: app/src/main/java/in/avimarine/boatangels/general/DialogUtils.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import android.view.View;
+import android.widget.Toast;
+import in.avimarine.boatangels.R;
+
+
+public class DialogUtils {
+
+    private static Toast toast;
+
+    public static void show(Context context, String message) {
```
Comment (Pazit) [inline app/src/main/java/in/avimarine/boatangels/general/DialogUtils.java]: code duplication. I would write it this way:

    public static void show(Context context, String message) {
		showByLength(context, message, Toast.LENGTH_SHORT)
	}

	public static void showLong(Context context, String message) {
		showByLength(context, message, Toast.LENGTH_LONG)
	}
	
 public static showByLength(Context context, String message, int toastLength)
 {
          if (message == null) {
             return;
         }
         if (toast == null && context != null) {
             toast = Toast.makeText(context, message, toastLength);
         }
         if (toast != null) {
             toast.setText(message);
             toast.show();
         }
 }

File: app/src/main/java/in/avimarine/boatangels/general/Versioning.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  }
+
+  public int getInstalledVersion() {
+    try {
+      PackageInfo pInfo = c.getPackageManager().getPackageInfo(c.getPackageName(), 0);
+      return pInfo.versionCode;
+    } catch (PackageManager.NameNotFoundException e) {
+      Log.d(TAG, "Error retreiving versioncode: ", e);
+    }
+    return -1;
```
Comment (Pazit) [inline app/src/main/java/in/avimarine/boatangels/general/Versioning.java]: Make sure there's no way to have a lower number of version saved in the DB, else, write a handling of this case.

File: app/src/main/java/in/avimarine/boatangels/activities/SplashActivity.java (hunk 1 above)
Comment (aayaffe) [reply app/src/main/java/in/avimarine/boatangels/activities/SplashActivity.java]: will add debug printing. 
I do not think this needs to be exposed to the lamen user.

File: app/src/main/java/in/avimarine/boatangels/fragments/MyBoatFragment.java (hunk 2 above)
Comment (aayaffe) [reply app/src/main/java/in/avimarine/boatangels/fragments/MyBoatFragment.java]: Done

File: app/src/main/java/in/avimarine/boatangels/general/DialogUtils.java (hunk 3 above)
Comment (aayaffe) [reply app/src/main/java/in/avimarine/boatangels/general/DialogUtils.java]: Fixed

File: app/src/main/java/in/avimarine/boatangels/general/Versioning.java (hunk 4 above)
Comment (aayaffe) [reply app/src/main/java/in/avimarine/boatangels/general/Versioning.java]: A very remote possibility, but I added a fix.

**Pull request closed without merging**